src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourceValidator.cs

```diff
@@ -0,0 +1,234 @@
+                SetAllowInsecureConnectionsProperty(packageSource);
+                EnsureValidSources(packageSource);
+            }
+            // If both name and source match, use the existing PackageSource.
+            else if (foundByName is not null && foundBySource is not null)
+            {
+                // Only the IsEnabled property could be changing.
+                foundByName.IsEnabled = isEnabled;
+                packageSource = foundByName;
+            }
+            // The source is changing on an existing PackageSource.
+            else if (foundByName is not null)
+            {
+                // Preserve existing properties by cloning the package source.
+                packageSource = new PackageSource(
+                    trimmedSource,
+                    foundByName.Name,
+                    isEnabled,
+                    foundByName.IsOfficial,
+                    foundByName.IsPersistable)
+                {
+                    IsMachineWide = foundByName.IsMachineWide,
+                    Credentials = foundByName.Credentials,
+                    ClientCertificates = foundByName.ClientCertificates,
+                    Description = foundByName.Description,
+                    ProtocolVersion = foundByName.ProtocolVersion,
+                    AllowInsecureConnections = foundByName.AllowInsecureConnections,
+                    DisableTLSCertificateValidation = foundByName.DisableTLSCertificateValidation,
+                    MaxHttpRequestsPerSource = foundByName.MaxHttpRequestsPerSource,
+                };
+            }
+            // The name is changing on an existing PackageSource.
+            else
+            {
+                // Preserve existing properties by cloning the package source.
+                packageSource = new PackageSource(
+                    foundBySource!.Source,
+                    trimmedName,
+                    isEnabled,
+                    foundBySource.IsOfficial,
+                    foundBySource.IsPersistable)
+                {
+                    IsMachineWide = foundBySource.IsMachineWide,
+                    Credentials = foundBySource.Credentials,
+                    ClientCertificates = foundBySource.ClientCertificates,
+                    Description = foundBySource.Description,
+                    ProtocolVersion = foundBySource.ProtocolVersion,
+                    AllowInsecureConnections = foundBySource.AllowInsecureConnections,
+                    DisableTLSCertificateValidation = foundBySource.DisableTLSCertificateValidation,
+                    MaxHttpRequestsPerSource = foundBySource.MaxHttpRequestsPerSource,
+                };
+            }
+
+            return packageSource;
+        }
+
+        /// <summary>
+        /// Validates the Uri of a remote or local package source.
+        /// The regex used here will eventually be supported in the Unified Settings registration.json file
```

Comment:
Left a hint and reminder that support is coming to expand our regex validation, resulting in a better user experience.

Reply:
Should we link to an issue in GitHub to make sure we also have this on our backlog radar and tie that to the code comment?

Reply:
Sure, added!